amida-tech/indaba-client
# Feature/inba 982 unassign task

#### What does this PR do?
This PR adds an "Unassign" functionality to the task options model in the PM's view of the  

#### Related JIRA tickets:
INBA-982
https://jira.amida.com/browse/INBA-982

#### How should this be manually tested?
Log into a PM user and navigate to a project that has several stages, a minimum of 

1) A completed stage
2) A stage that work has begun on
3) A stage that has no work but has been assigned.
4) A stage you create at the end of the project for this test. 

Perform the following tests on each of the above stages in the task options menu:

A) Open the task options menu.
B) Attempt to click on the "unassign" option presented.
C) Click "SAVE" in the task options modal.
D) Click refresh or F5 to verify that the data performs as expected upon refresh. 

**Expected results:**
1)
     A) The modal should open, and be greyed out.
     B) You should be unable to select the unassign function. 
     C) Saving should result in no change of the data presented

2)
     A) The modal should open, and be greyed out.
     B) You should be unable to select the unassign function. 
     C) Saving should result in no change of the data presented

3)
     A) The modal should open, and allow selection
     B) You should be able to select the unassign function. 
     C) Saving should result in the removal of the stage assignment from the chosen stage.

4)
     A) The modal should open, and allow selection
     B) You should be able to select the unassign function. 
     C) Saving should result in the removal of the stage assignment from the chosen stage.

Please also unassign a stage, then assign it to another user, and then refresh the page making sure that the reassignment has worked.
